File: Ch02/02_03/median.go
// Calculate the median of a slice of numbers
package main

import (
	"fmt"
	"sort"
)

func main() {
	nums := []float64{2, 1, 3}

	sort.Float64s(nums)
	var median float64
	i := len(nums) / 2
	if len(nums)%2 == 1 {
		median = nums[i]
	} else {
		median = (nums[i-1] + nums[i]) / 2
	}
	fmt.Println(median)
}

File: Ch02/02_03/median.py
# Calculate the median of a list of numbers


numbers = [2, 1, 3]

numbers.sort()
i = len(numbers) // 2
if len(numbers) % 2 == 1:
    median = numbers[i]
else:
    median = (numbers[i-1] + numbers[i]) / 2
print(median)
